pkg/controller/controller_test.go

```diff
@@ -189,6 +224,158 @@ func newBPWithOutputArtifact() *crv1alpha1.Blueprint {
 	}
 }
 
+func newBPWithDeferPhaseAndErrInDeferPhase() *crv1alpha1.Blueprint {
+	return &crv1alpha1.Blueprint{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "test-blueprint-defer-defererr-",
+		},
+		Actions: map[string]*crv1alpha1.BlueprintAction{
+			"backup": {
+				OutputArtifacts: map[string]crv1alpha1.Artifact{
+					"opArtPhaseOne": {
+						KeyValue: map[string]string{
+							"op": "{{ .Phases.backupPhaseOne.Output.value }}",
+						},
+					},
+					"opArtPhaseTwo": {
+						KeyValue: map[string]string{
+							"op": "{{ .Phases.backupPhaseTwo.Output.value }}",
+						},
+					},
+				},
+				Phases: []crv1alpha1.BlueprintPhase{
+					backupPhase("backupPhaseOne", "mainValue"),
+					backupPhase("backupPhaseTwo", "mainValueTwo"),
+				},
+				DeferPhase: &crv1alpha1.BlueprintPhase{
+					Name: "deferPhase",
+					Func: function.KubeTaskFuncName,
+					Args: map[string]interface{}{
+						"image":     mysqlSidecarImage,
+						"namespace": "default",
+						"command":   []string{"exit", "1"},
+					},
+				},
+			},
+		},
+	}
+}
+
+func newBPWithDeferPhaseAndErrInCorePhase() *crv1alpha1.Blueprint {
+	return &crv1alpha1.Blueprint{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "test-blueprint-defer-coreerr-",
+		},
+		Actions: map[string]*crv1alpha1.BlueprintAction{
+			"backup": {
+				OutputArtifacts: map[string]crv1alpha1.Artifact{
+					"opArtPhaesOne": {
+						KeyValue: map[string]string{
+							"op": "{{ .Phases.backupPhaseOne.Output.value }}",
+						},
+					},
+					"opArtDeferPhase": {
+						KeyValue: map[string]string{
+							"op": "{{ .DeferPhase.Output.value }}",
+						},
+					},
+				},
+				Phases: []crv1alpha1.BlueprintPhase{
+					backupPhase("backupPhaseOne", "mainValue"),
+					{
+						Name: "backupPhaseTwo",
+						Func: function.KubeTaskFuncName,
+						Args: map[string]interface{}{
+							"image":     mysqlSidecarImage,
+							"namespace": "default",
+							"command":   []string{"exit", "1"},
+						},
+					},
+				},
+				DeferPhase: deferPhase(),
+			},
+		},
+	}
+}
+
+func backupPhase(name, outputArtValue string) crv1alpha1.BlueprintPhase {
+	return crv1alpha1.BlueprintPhase{
+		Name: name,
+		Func: function.KubeTaskFuncName,
+		Args: map[string]interface{}{
+			"image":     mysqlSidecarImage,
+			"namespace": "default",
+			"command":   []string{"kando", "output", "value", outputArtValue},
+		},
+	}
+}
+
+func deferPhase() *crv1alpha1.BlueprintPhase {
+	return &crv1alpha1.BlueprintPhase{
+		Name: "deferPhase",
+		Func: function.KubeTaskFuncName,
+		Args: map[string]interface{}{
+			"image":     mysqlSidecarImage,
+			"namespace": "default",
+			"command":   []string{"kando", "output", "value", "deferValue"},
+		},
+	}
+}
```

Comment:
Feels like this is almost the same as `backupPhase()`. Can we just change the `backupPhase()` signature to something like:

```go
func backupPhase(name string, command []string) crv1alpha1.BlueprintPhase
```